### PR TITLE
Bugfix/agency lock left behind

### PR DESCRIPTION
**Pull request by kvahed**

### CODE REVIEW

File: arangod/Cluster/ClusterInfo.cpp [hunk 1]
```diff
@@ -4430,8 +4430,9 @@ arangodb::Result ClusterInfo::agencyHotBackupLock(std::string const& backupId,
 
   if (!rv->slice().isObject() || !rv->slice().hasKey("results") ||
       !rv->slice().get("results").isArray() || rv->slice().get("results").length() != 2) {
-    return arangodb::Result(TRI_ERROR_HOT_BACKUP_INTERNAL,
-                            "invalid agency result while acuiring backup lock");
+    return arangodb::Result(
+      TRI_ERROR_HOT_BACKUP_INTERNAL,
+      "invalid agency result while acuiring backup lock");
```
Comment (jsteemann) [inline arangod/Cluster/ClusterInfo.cpp]:
```suggestion
      "invalid agency result while acquiring backup lock");
```

File: arangod/Cluster/ClusterInfo.cpp [hunk 2]
```diff
@@ -4555,11 +4558,10 @@ arangodb::Result ClusterInfo::agencyHotBackupUnlock(std::string const& backupId,
...
       if (!result.slice().isArray() || result.slice().length() != 1 ||
           !result.slice()[0].hasKey(modepv) || !result.slice()[0].get(modepv).isString()) {
-        return arangodb::Result(TRI_ERROR_HOT_BACKUP_INTERNAL, std::
-                                                                       string("invalid JSON from agency, when desctivating supervision mode:") +
-                                                                   result
-                                                                       .slice()
-                                                                       .toJson());
+        return arangodb::Result(
+          TRI_ERROR_HOT_BACKUP_INTERNAL,
+          std::string("invalid JSON from agency, when desctivating supervision mode:") +
```
Comment (jsteemann) [inline arangod/Cluster/ClusterInfo.cpp]:
```suggestion
          std::string("invalid JSON from agency, when deactivating supervision mode:") +
```

File: arangod/Cluster/ClusterMethods.cpp [hunk 3]
```diff
@@ -4026,6 +4026,13 @@ arangodb::Result hotBackupCoordinator(VPackSlice const payload, VPackBuilder& re
     double timeout = (payload.isObject() && payload.hasKey("timeout"))
                          ? payload.get("timeout").getNumber<double>()
                          : 120.;
+    // unreasonably short even under allowInconsistent
+    if (timeout < 2.5) {
```
[the remark this answers is not in the document]
Comment (kvahed) [reply arangod/Cluster/ClusterMethods.cpp]: changed as discussed

File: CHANGELOG [hunk 4]
```diff
@@ -1,7 +1,9 @@
 devel
 -----
 
-* Remove operations documents in the cluster will now use an optimization, if all sharding keys 
+* Hot bckup agency lock could be left behind when short timeout was hit
```
Comment (neunhoef) [inline CHANGELOG]: Let's remove this, too.

**Comment by kvahed** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/6298/

**Pull request merged**